chain/epoch-manager/src/lib.rs

```diff
@@ -200,36 +190,95 @@ impl EpochManager {
     }
 
     pub fn new_arc_handle(store: Store, genesis_config: &GenesisConfig) -> Arc<EpochManagerHandle> {
-        Self::new_arc_handle_with_test_overrides(store, genesis_config, None)
+        let chain_id = genesis_config.chain_id.as_str();
+        if chain_id == near_primitives::chains::MAINNET
+            || chain_id == near_primitives::chains::TESTNET
+        {
+            let epoch_config_store = EpochConfigStore::for_chain_id(chain_id).unwrap();
+            return Self::new_arc_handle_from_epoch_config_store(
+                store,
+                genesis_config,
+                epoch_config_store,
+            );
+        }
+
+        let epoch_config = if chain_id.starts_with("test-chain-") {
+            // We still do this for localnet as nayduck depends on it.
```

Comment:
This particular block of code won't impact mainnet, because its chain name doesn't start from `test-chain-`, right?

For mocknet, I have two ideas:
1. Change constructor every time we introduce new mocknet. Always derive base config from mainnet config of the latest version.
2. Add new folder to the `epoch_configs/` corresponding to each separate mocknet instance. Shouldn't take too much place in repo, because we need to support only latest mainnet and couple protocol transitions on top of that.

Reply:
I would vote for option 1: use mainnet latest config as the as base config and update it based on the specific mocknet setup (eg. num seats). otherwise, we will keep updating the mocknet in epoch_configs, which should be based on mainnet with fixed set of overrides anyways.